### PR TITLE
Remove exception logging for ItemNotReadyError

### PR DESCRIPTION
**Pull request by nickzuber**
Attempting to reach an item that isn't ready is basically a 404 and there isn't really a need to log anything to the master in this case.

Also when attempting to reach an artifact that isn't ready, the `clusterrunner build` command will continuously try to request it until the client is explicitly stopped -- this will spam the master logs.

Resolves: #342

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @nickzuber has signed the CLA. Thanks for the pull request!

File: app/web_framework/cluster_base_handler.py [hunk 1]
```diff
@@ -38,8 +38,10 @@ def _handle_request_exception(self, ex):
         :param ex: The exception that was caught
         :type ex: Exception
         """
-        # _handle_request_exception() is called in the exception handler, so we can still use logger.exception.
-        self._logger.exception('Exception occurred during request to {}.', self.request.uri)
+        # Requests to access an item that isn't ready shouldn't log anything
+        if type(ex) is not ItemNotReadyError:
```
Comment (josephharrington) [inline app/web_framework/cluster_base_handler.py]: While you're here, do you think there are any other exception types we should remove from logging? 404s seem like they'd be another good one to omit. Thoughts?

File: app/web_framework/cluster_base_handler.py [hunk 2]
```diff
@@ -38,8 +38,10 @@ def _handle_request_exception(self, ex):
         :param ex: The exception that was caught
         :type ex: Exception
         """
-        # _handle_request_exception() is called in the exception handler, so we can still use logger.exception.
-        self._logger.exception('Exception occurred during request to {}.', self.request.uri)
+        # Requests to access an item that isn't ready shouldn't log anything
+        if not isinstance(ex, ItemNotReadyError):
```
Comment (josephharrington) [inline app/web_framework/cluster_base_handler.py]: While you're here, do you think there are any other exception types we should remove from logging? 404s seem like they'd be another good one to omit. Thoughts?

Comment (nickzuber) [inline app/web_framework/cluster_base_handler.py]: Good idea, that could be here as well. The other types of exceptions seem like they'd be important to have in the logs, so do we just want to `ItemNotReadyError` and `ItemNotFoundError`?

Comment (josephharrington) [inline app/web_framework/cluster_base_handler.py]: Sounds good to me. :)

**Pull request merged**